invenio_rdm_records/services/schemas/access.py

```diff
@@ -33,26 +31,22 @@ class Agent(Schema):
     user = fields.Integer(required=True)
 
 
+class EmbargoSchema(Schema):
+    """Schema for an embargo on the record."""
+
+    # TODO: check that ('until' is in the future <=> 'active' == true)
+    active = fields.Bool(required=True)
+    until = ISODateString(required=True)
+    reason = SanitizedUnicode(required=True)
```

Comment:
alright, then we'd have:
`reason` being purely optional
`until` being required if `active == True` (in which case it must be set to some future date, too)
`active` being required if `until` isn't set (in which case, `active` can only have the value `False`)
also, `active`'s value must be `False` if `until` lies in the past

i was also thinking about having the entire `embargo` structure being optional -- if there's no embargo in the first place, it can simply be left out?
i think, that's effectively the same as an `embargo` field which only has the `active` property set to `False`, and nothing else...
what do you think?

Reply:
> reason being purely optional
:heavy_check_mark: 

> until being required if active == True (in which case it must be set to some future date, too)
:heavy_check_mark: 

> active being required if until isn't set 
yes - but a one flow of conditions is perhaps easier (given `until` above depends on `active` we don't want `active` to depend on `until`):
`active` can be true or false. If false, `until` is empty or in the past. If true, `until` is in the future. `reason` is alwasy optional.

> the entire embargo structure being optional
yes and the current Schema allows that, so we are already good.